monkey/monkey_island/cc/resources/configuration_import.py

```diff
@@ -0,0 +1,94 @@
+import json
+import logging
+from dataclasses import dataclass
+from json.decoder import JSONDecodeError
+
+import flask_restful
+from flask import request
+
+from common.utils.exceptions import InvalidConfigurationError
+from monkey_island.cc.resources.auth.auth import jwt_required
+from monkey_island.cc.services.config import ConfigService
+from monkey_island.cc.services.utils.config_encryption import (
+    InvalidCredentialsError,
+    decrypt_config,
+    is_encrypted,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class ImportStatuses:
+    UNSAFE_OPTION_VERIFICATION_REQUIRED = "unsafe_options_verification_required"
+    INVALID_CONFIGURATION = "invalid_configuration"
+    INVALID_CREDENTIALS = "invalid_credentials"
+    IMPORTED = "imported"
+
+
+@dataclass
+class ResponseContents:
+    import_status: str = ImportStatuses.IMPORTED
+    message: str = ""
+    status_code: int = 200
+    config: str = ""
+    config_schema: str = ""
+
+    def form_response(self):
+        return self.__dict__
+
+
+class ConfigurationImport(flask_restful.Resource):
+    SUCCESS = False
+
+    @jwt_required
+    def post(self):
+        request_contents = json.loads(request.data)
+        try:
+            config = ConfigurationImport._get_plaintext_config_from_request(request_contents)
+            if request_contents["unsafeOptionsVerified"]:
+                ConfigurationImport.import_config(config)
+                return ResponseContents().form_response()
+            else:
+                return ResponseContents(
+                    config=json.dumps(config),
+                    config_schema=ConfigService.get_config_schema(),
+                    import_status=ImportStatuses.UNSAFE_OPTION_VERIFICATION_REQUIRED,
+                ).form_response()
+        except InvalidCredentialsError:
+            return ResponseContents(
+                import_status=ImportStatuses.INVALID_CREDENTIALS,
+                message="Invalid credentials provided",
+            ).form_response()
+        except InvalidConfigurationError:
+            return ResponseContents(
+                import_status=ImportStatuses.INVALID_CONFIGURATION,
+                message="Invalid configuration supplied. "
+                "Maybe the format is outdated or the file has been corrupted.",
+            ).form_response()
+
+    @staticmethod
+    def _get_plaintext_config_from_request(request_contents: dict) -> dict:
+        if ConfigurationImport.is_config_encrypted(request_contents["config"]):
+            return decrypt_config(request_contents["config"], request_contents["password"])
```

Comment:
`decrypt_config()` should do exactly that and only that, but now it also tries to parse the config by calling `json.load()`. If this call throws a JsonDecodeError, nothing catches it and it isn't fed back to the user. It also duplicates the `json.load()` functionality below on line 75. What about something like:

```python
config = request_contents["config"]
if is_config_encrypted(config):
  config = decrypt_config(config)

try:
  return json.loads(config)
...
```